tests/suites/test_suite_ccm.function

```diff
@@ -591,6 +591,41 @@ exit:
 }
 /* END_CASE */
 
+/* Provide incomplete auth data on first update_ad.
+ * Provide too much auth data on second update_ad */
+/* BEGIN_CASE */
+void mbedtls_ccm_incomplete_ad_and_overflow( int cipher_id, int mode,
+                                             data_t * key, data_t * iv,
+                                             data_t * add )
+{
+    mbedtls_ccm_context ctx;
+
+    /* New auth buffer containing same data as original one,
+     * with added extra byte at the end */
+    uint8_t* add_extended = NULL;
+    ASSERT_ALLOC( add_extended, add->len + 1 );
```

Comment:
I think it would be simpler to pass `add->len - 1` bytes from `add` first, then pass 2 bytes from a local 2-byte array.

Also goes for `mbedtls_ccm_incomplete_update_overflow`.